submit/controllers/ui/new/classification.py

```diff
@@ -127,24 +126,17 @@ def classification(method: str, params: MultiDict, session: Session,
         'form': form
     }
 
-    if method == 'POST':
-        if not form.validate():
-            raise BadRequest(response_data)
-
-        command = SetPrimaryClassification(category=form.category.data,
-                                           creator=submitter, client=client)
-        if not validate_command(form, command, submission, 'category'):
-            raise BadRequest(response_data)
-
+    command = SetPrimaryClassification(category=form.category.data,
+                                       creator=submitter, client=client)
+    if method == 'POST' and form.validate()\
+       and validate_command(form, command, submission, 'category'):
         try:
             submission, _ = save(command, submission_id=submission_id)
             response_data['submission'] = submission
-        except SaveError as e:
-            logger.error('Could not save command: %s', e)
-            raise InternalServerError(response_data) from e
+        except SaveError as ex:
```

Comment:
I'm curious what this error actually triggers. It seems that if the application can't save the form values that there could be a serious problem with the application. I guess I'm wondering if certain errors should impact the 'healthy' status of the submission UI. This points to application-wide prioritizing and handling of critical errors and obviously goes beyond the code here.

Reply:
The InternalServerError gets handled by the Flask error handlers.